src/mame/matsushita/kn5000.cpp

```diff
@@ -591,6 +618,16 @@ void kn5000_state::machine_start()
 	save_item(NAME(m_mstat));
 	save_item(NAME(m_sstat));
 
+	if(m_extension)
+	{
+		m_extension->rom_map(m_extension_view[1], 0x280000, 0x2fffff);
+		m_extension_view.select(1);
+	}
+	else
+	{
+		m_extension_view.select(0);
+	}
```

Comment:
This looks like you’re using views inappropriately.  The whole point of using a view is so you don’t have to install handlers on-the-fly.  If you’re installing handlers when memory mapping changes, you’re doing it wrong.  You should make the extension device set up whatever it needs to once at start.